performance/shared/src/main/kotlin/org/jetbrains/benchmarksLauncher/launcher.kt

```diff
@@ -19,21 +19,38 @@ package org.jetbrains.benchmarksLauncher
 import kotlin.math.sqrt
 import org.jetbrains.report.BenchmarkResult
 import org.jetbrains.kliopt.*
-import kotlin.reflect.KFunction0
 
-abstract class Launcher(val numWarmIterations: Int, val numberOfAttempts: Int, val prefix: String = "") {
-    class Results(val mean: Double, val variance: Double)
 
+abstract class Launcher {
     abstract val benchmarks: BenchmarksCollection
 
-    fun add(name: String, benchmark:() -> Any?) {
-        fun benchmarkWrapper() {
-            benchmark()
+    fun add(name: String, benchmark: BenchmarkEntry) {
+        benchmarks[name] = benchmark
+    }
+
+    fun runBenchmark(benchmarkInstance: Any?, benchmark: BenchmarkEntry, repeatNumber: Int): Long {
+        var i = repeatNumber
+        return if (benchmark is InstanceBenchmarkEntry) {
+            cleanup()
+            measureNanoTime {
+                while (i-- > 0) benchmark.lambda(benchmarkInstance!!)
```

Comment:
Maybe move `if (benchmark is InstanceBenchmarkEntry)` there?

Reply:
I made this  to avoid mostly all unconnected actions to make measurement as accurate as possible